velox/connectors/hive/SplitReader.cpp

```diff
@@ -197,6 +222,15 @@ std::vector<TypePtr> SplitReader::adaptColumns(
             INTEGER(),
             velox::variant(hiveSplit_->tableBucketNumber.value()));
       }
+    } else if (infoColumn != hiveSplit_->infoColumns.end()) {
+      auto infoColumnType =
+          readerOutputType_->childAt(readerOutputType_->getChildIdx(fieldName));
+      auto value = VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
```

Comment:
@aditi-pandit I think it is a good idea

Reply:
@majetideepak @Yuhta : My understanding is that we are trying to avoid the use of variant in non-test code. So adding that are a part of Velox API seems a bit odd to me.

@mbasmanova : Any thoughts ?

Reply:
We are using a variant here in the non-test code.
It is also inefficient since there is another switch inside ConstantVector to extract the variant value.
How about we pass a `constantValue` then?

Reply:
I created a PR here that removes the existing usage of variant https://github.com/facebookincubator/velox/pull/8929

Reply:
@majetideepak : Agree that translating directly from std::string to ConstantVector within SplitReader is a better solution than passing those from the application since there was some HiveSplit specific parsing to take care here.

The variant use is also removed. 

Thanks for your PR that fixes this in all the places that was used. Its better to do the cleanup PR separately.

I'll update this code once your PR is submit.

Reply:
Rebased code.